va/gsb_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2014 ISRG.  All rights reserved
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package va
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/cactus/go-statsd-client/statsd"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/golang/mock/gomock"
+	"github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/jmhodges/clock"
+	safebrowsing "github.com/letsencrypt/boulder/Godeps/_workspace/src/github.com/letsencrypt/go-safe-browsing-api"
+	"github.com/letsencrypt/boulder/core"
+)
+
+func TestIsSafeDomain(t *testing.T) {
+	// TODO(jmhodges): use more of the GSB lib by teaching it how to not make http requests
+	stats, _ := statsd.NewNoopClient()
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	sbc := NewMockSafeBrowsing(ctrl)
+	sbc.EXPECT().IsListed("good.com").Return("", nil)
+	sbc.EXPECT().IsListed("bad.com").Return("bad", nil)
+	sbc.EXPECT().IsListed("errorful.com").Return("", errors.New("welp"))
+	sbc.EXPECT().IsListed("outofdate.com").Return("", safebrowsing.ErrOutOfDateHashes)
+	va := NewValidationAuthorityImpl(&PortConfig{}, sbc, stats, clock.NewFake())
+
+	resp, err := va.IsSafeDomain(&core.IsSafeDomainRequest{"good.com"})
+	if err != nil {
+		t.Errorf("good.com: want no error, got '%s'", err)
+	}
+	if !resp.IsSafe {
+		t.Errorf("good.com: want true, got %t", resp.IsSafe)
+	}
+	resp, err = va.IsSafeDomain(&core.IsSafeDomainRequest{"bad.com"})
+	if err != nil {
+		t.Errorf("bad.com: want no error, got '%s'", err)
+	}
+	if resp.IsSafe {
+		t.Errorf("bad.com: want false, got %t", resp.IsSafe)
+	}
+	_, err = va.IsSafeDomain(&core.IsSafeDomainRequest{"errorful.com"})
+	if err == nil {
+		t.Errorf("errorful.com: want error, got none")
+	}
+	resp, err = va.IsSafeDomain(&core.IsSafeDomainRequest{"outofdate.com"})
+	if err != nil {
+		t.Errorf("outofdate.com: want no error, got '%s'", err)
+	}
+	if !resp.IsSafe {
+		t.Errorf("outofdate.com: IsSafeDomain should fail open on out of date hashes")
+	}
+}
+
+func TestAllowNilInIsSafeDomain(t *testing.T) {
+
+	// This is mocked out at the wrong level because the gsb lib we rely on is a
```

Comment:
The interface being at the IsListed method instead of deeper in the safe browsing library's SafeBrowsing object (like, maybe, the code that makes HTTP requests).